### ISSUE TITLE
googler no results

### ISSUE THREAD

**Issue by mtv111** [jarun/googler#299]
googler is currently producing "No results" for any searches

**Comment by raffaele96x** [comment]
I have the same problem, on windows 10 and on linux..
Any solution? Thanks

**Comment by jarun** [comment]
Yes, I can see this happening intermittently. A few queries fetch results, most fail.

**Comment by jarun** [comment]
@zmwangx please take a look.

**Comment by jarun** [comment]
Here's what I get in debug mode:

```
$ ./googler hello world --debug 
[DEBUG] googler version 3.9
[DEBUG] Python version 3.6.8
[DEBUG] Connecting to new host www.google.com
[DEBUG] Fetching URL /search?ie=UTF-8&oe=UTF-8&q=hello+world&sei=3kRpjAL4EeqwWphUGwqfFQ
[DEBUG] Cookie: 1P_JAR=2019-11-09-13
[DEBUG] Response body written to '/tmp/googler-response-xikqk0vv.html'.
No results.
googler (? for help) 
googler (? for help)
```

Response attached. Looks like the results are there but the format might have changed.


[googler-response.zip](https://github.com/jarun/googler/files/3827359/googler-response.zip)

**Comment by mengxuzhao** [comment]
I'm having exactly the same problem. Tried the latest version of googler in OSX, ubuntu and fedora.

**Comment by jessestuart** [comment]
Did a bit of digging and found the cause; still looking into a solution.

TL;DR I believe the user-agent `googler` is using seems to be returning an alternate version of the site, which lacks the tags/classnames `googler` is looking for to scrape.

Specifically, [the parsing begins by selecting all `div.g` elements](https://github.com/jarun/googler/blob/05d51875022cd6a6a1a6430b606df89dcfec9400/googler#L2112). This matches the HTML I see when performing an in-browser search:

<img width="50%" alt="Screen Shot 2019-11-09 at 11 44 24 PM" src="https://user-images.githubusercontent.com/583147/68539070-d50c3900-034b-11ea-8a96-14e14fff41e4.png"><img width="50%" alt="Screen Shot 2019-11-09 at 11 44 41 PM" src="https://user-images.githubusercontent.com/583147/68539073-d89fc000-034b-11ea-9b1e-3d087876ccad.png">

--------------------------------

However, if I disable Javascript, I get a result that matches what's returned by `googler` when using the `--debug` flag — the markup seems to be using any variation of CSS-in-js, based on the mangled classnames. Whatever the case, there are no matches for the `div.g` selector, and thus the CLI returns no results:

<img width="50%" alt="Screen Shot 2019-11-09 at 11 43 45 PM" src="https://user-images.githubusercontent.com/583147/68539121-9a56d080-034c-11ea-9747-a6e2258c4242.png"><img width="50%" alt="Screen Shot 2019-11-09 at 11 43 07 PM" src="https://user-images.githubusercontent.com/583147/68539118-8c08b480-034c-11ea-9b5d-0cad61382cec.png">

**Comment by zmwangx** [comment]
Is this still an issue? I can't reproduce. Sometimes Google do random experiments and we can only wait it out.

**Comment by jarun** [comment]
Yes, I am unable to repro too.

**Comment by roberdam** [comment]
Is working now, yesterday was giving no results

**Comment by ugjka** [comment]
Was not working and now working again

**Comment by jessestuart** [comment]
👍 Yup, working for me as well now too — the `request` module is again returning the expected format. `¯\_(ツ)_/¯`

@mtv111 Care to close this issue, or are you still having issues?

**Comment by mtv111** [comment]
> 👍 Yup, working for me as well now too — the `request` module is again returning the expected format. `¯\_(ツ)_/¯`
> 
> @mtv111 Care to close this issue, or are you still having issues?

Yea seems to be working now. I’ll close.

**Comment by taewookim** [comment]
still not working for me

**Comment by jarun** [comment]
Please use the latest available release.

**Comment by mengxuzhao** [comment]
This issue seems resolved. Thank you.

**Comment by zmwangx** [comment]
For future reference: if you landed here through search, please check the wiki article [*What to do when you encounter "No results"*](https://github.com/jarun/googler/wiki/What-to-do-when-you-encounter-%22No-results%22). A new variant should go into a new issue, so I'm locking this one.
